### PR TITLE
docs: Change docs to `sphinx` and use rST docstring format

### PR DESCRIPTION
**Pull request by an-dyy**
This PR switches docs to `sphinx` from `mkdocs` and change docstring format to `reStructedText`

### CODE REVIEW

File: docs/rtd_requirements.txt [hunk 1]
```diff
@@ -3,3 +3,5 @@ mkdocs >="1.2.2"
 mkdocs-material >= "7.3.2"
```
Comment (anand2312) [inline docs/rtd_requirements.txt:3]: Not needed anymore

**Review by anand2312 — CHANGES_REQUESTED** [review]
I think you've missed the CI file that auto-deploys docs (but I'm not sure I'm doing this on mobile 😔)

File: docs/rtd_requirements.txt (hunk 1 above)
Comment (an-dyy) [reply docs/rtd_requirements.txt]: Thanks, fixed

**Comment by an-dyy** [comment]
Merging right now, instead of finishing then merging to save some time later

**Pull request merged**